pyiron/lammps/control.py

```diff
@@ -789,6 +789,58 @@ def energy_pot_per_atom(self):
             self['dump___1'] += ' c_energy_pot_per_atom'
             self['dump_modify___1'] = self['dump_modify___1'][:-1] + ' %20.15g"'
 
+    def _set_group_by_id(self, group_name, ids):
+        if len(ids) < 1:
+            raise ValueError('Group ids must have at least length one, but got {}'.format(ids))
+        if np.any([isinstance(id_, bool) or not isinstance(id_, (int, np.int64)) for id_ in ids]):
+            # Note: it turns out bool is a subclass of int. Weird, eh.
+            raise TypeError('Group ids must be integers, but got {}'.format(ids))
+        if np.any(np.array(ids) < 0):
+            raise ValueError('Group ids must be non-negative to be parsed by Lammps, but got {}'.format(ids))
+        self['group___{}'.format(group_name)] = 'id {}'.format(' '.join(np.array(ids).astype(int).astype(str)))
+
+    def _fix_to_three_vector(self, ids, vector, fix_string, conversion):
+        if len(vector) != 3:
+            raise ValueError('{} must have three components, but got {}'.format(fix_string, vector))
+        vector = list(vector)
+        for i, v in enumerate(vector):
+            if v is None:
+                vector[i] = 'NULL'
+            else:
+                vector[i] = str(v * conversion)
+        name = str(hash(tuple(ids))).replace('-', 'm')  # A unique name for the group
+        self._set_group_by_id(name, ids)
+        self['fix___{}_{}'.format(fix_string.replace(' ', '_'), name)] = '{} {} {}'.format(
+            name, fix_string, ' '.join(vector)
+        )
+
+    def fix_move_linear_by_id(self, ids, velocity):
+        """
+        Displace atoms at each timestep. Creates a new group with a unique name based off the hash of the ids.
+
+        Args:
+            ids (list/numpy.ndarray): Integer ids of the atoms to move in the job's structure.
+            velocity (list/numpy.ndarray/tuple): The velocity in x-y-z-direction for the group. `None` arguments are
+                converted to Lammps 'NULL' values and the velocity in this direction is left unchanged.
```

Comment:
Should that at least get a warning that `velocity` has to be rotated if the simulation box is rotated?

Reply:
The problem is that at the `LammpsControl` level we have no access to whether or not the box is rotated; this only comes in when the rotation matrix is explicitly passed. We could certainly make this general in the future (and then allow accessing this power directly at the job level) but right now this command knows nothing about it.

Reply:
I meant a warning in the docstring to reduce the surprise.

Reply:
Ah -- You bet!